qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -40,20 +59,81 @@ def new_prep_template_get_req(study_id):
     data_types = sorted(Study.all_data_types())
 
     # Get all the ENA terms for the investigation type
-    ontology = Ontology(convert_to_id('ENA', 'ontology'))
-    ena_terms = sorted(ontology.terms)
-    # make "Other" last on the list
-    ena_terms.remove('Other')
-    ena_terms.append('Other')
+    ontology_info = _get_ENA_ontology()
 
-    ontology_info = {'ENA': ena_terms,
-                     'User': sorted(ontology.user_defined_terms)}
     return {'status': 'success',
             'prep_files': prep_files,
             'data_types': data_types,
             'ontology': ontology_info}
 
 
+def prep_template_ajax_get_req(prep_id):
+    """Returns the prep tempalte information needed for the AJAX handler
+
+    Parameters
+    ----------
+    prep_id : int
+        The prep template id
+
+    Returns
+    -------
+    dict of {str: object}
+        A dictionary with the following keys:
+        - status: str, whether if the request is successful or not
+        - message: str, if the request is unsuccessful, a human readable error
+        - name: str, the name of the prep template
+        - files: list of str, the files available to update the prep template
+        - download_prep: int, the filepath_id of the prep file
+        - download_qiime, int, the filepath_id of the qiime mapping file
+        - num_samples: int, the number of samples present in the template
+        - num_columns: int, the number of columns present in the template
+        - investigation_type: str, the investigation type of the template
+        - ontology: str, dict of {str, list of str} conataining the information
+        of the ENA ontology
+        - artifact_attached: bool, whethe the template has an artifact attached
+        - study_id: int, the study id of the template
+    """
+    # Currently there is no name attribute, but it will be soon
+    name = "Prep information %d" % prep_id
+    pt = PrepTemplate(prep_id)
+    artifact_attached = pt.artifact is not None
+    study_id = pt.study_id
+    files = [f for _, f in get_files_from_uploads_folders(study_id)
+             if f.endswith(('txt', 'tsv'))]
+
+    # The call to list is needed because keys is an iterator
+    num_samples = len(list(pt.keys()))
+    num_columns = len(pt.categories())
+    investigation_type = pt.investigation_type
+
+    # Retrieve the information to download the prep template and QIIME
+    # mapping file. See issue https://github.com/biocore/qiita/issues/1675
+    download_prep = []
+    download_qiime = []
+    for fp_id, fp in pt.get_filepaths():
+        if 'qiime' in basename(fp):
+            download_qiime.append(fp_id)
+        else:
+            download_prep.append(fp_id)
+    download_prep = download_prep[0]
+    download_qiime = download_qiime[0]
+
+    ontology = _get_ENA_ontology()
+
+    return {'status': 'success',
```

Comment:
For consistency with the other similar calls. I agree, we should remove it and I think it will go away once we start discussing the REST api in depth, but for now this is the best in terms of consistency.